authz/sdk_end2end_test.go

```diff
@@ -0,0 +1,164 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package authz
+
+import (
+	"context"
+	"net"
+	"testing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	pb "google.golang.org/grpc/test/grpc_testing"
+)
+
+type testServer struct {
+	pb.UnimplementedTestServiceServer
+}
+
+func (s *testServer) UnaryCall(ctx context.Context, req *pb.SimpleRequest) (*pb.SimpleResponse, error) {
+	return &pb.SimpleResponse{}, nil
+}
+
+func startServer(t *testing.T, policy string) string {
+	i, _ := NewStatic(policy)
+	serverOpts := []grpc.ServerOption{
+		grpc.ChainUnaryInterceptor(i.UnaryInterceptor),
+		grpc.ChainStreamInterceptor(i.StreamInterceptor),
+	}
+	lis, err := net.Listen("tcp", ":0")
+	if err != nil {
+		t.Fatalf("error listening: %v", err)
```

Comment:
Avoid passing `t` around if possible, or maybe declare `t.Helper` if you need to.  Instead, it's nicer to return an `error` from this and put the `t.Fatal` in the test case main function directly.

Reply:
Done. I moved the function body into test case.